File: CodingBat/Python/List-2/sum13.py
'''
Return the sum of the numbers in the array, returning 0 for an empty array.
Except the number 13 is very unlucky, so it does not count and numbers that
come immediately after a 13 also do not count.
'''

def sum13(nums):
    nums.append(0)
    count = 0

    for i in range(len(nums)):
        if nums[i] == 13 and nums[i + 1] != 13:
            count -= (nums[i] + nums[i + 1])
        elif nums[i] == 13:
            count -= nums[i]

        count += nums[i]

    return count

File: CodingBat/Java/Array-2/Sum13.java
/**
 * Return the sum of the numbers in the array, returning 0 for an empty array.
 * Except the number 13 is very unlucky, so it does not count and numbers that
 * come immediately after a 13 also do not count.
 */

public class Sum13 {
    public int sum13(int[] nums) {
        int c = 0;

        if (nums.length == 0)
            return 0;

        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] != 13)
                c += nums[i];
            else if (nums[i] == 13 && nums[i + 1] != 13)
                c -= nums[i + 1];
        }

        if (nums[nums.length - 1] != 13)
            c += nums[nums.length - 1];

        return c;
    }
}
